storage/raft.go

```diff
@@ -32,8 +33,16 @@ type committedCommand struct {
 	cmd      proto.InternalRaftCommand
 }
 
-// raft is the interface exposed by a raft implementation.
-type raft interface {
+// raftInterface is the interface exposed by a raft implementation.
+type raftInterface interface {
+	// createGroup initializes a raft group with the given id.
+	createGroup(int64) error
+
+	removeGroup(int64) error
```

Comment:
No comment for removeGroup? Probably should add one for symmetry, if nothing else.

Comment for restoreGroup is not that informative. Maybe include a sentence about when it's called. I imagine during startup when the disk is being scanned? We're probably going to want to remove the range metadata from the local key space and instead rely solely on the group's on-disk state. Maybe add a TODO to that effect.